sql/src/main/java/io/crate/metadata/ReferenceInfos.java

```diff
@@ -131,22 +132,21 @@ public void clusterChanged(ClusterChangedEvent event) {
         }
 
         Set<String> newCurrentSchemas = getNewCurrentSchemas(event.state().metaData());
-        synchronized (schemas) {
-            Sets.SetView<String> nonBuiltInSchemas = Sets.difference(schemas.keySet(), builtInSchemas.keySet());
-            Set<String> deleted = Sets.difference(nonBuiltInSchemas, newCurrentSchemas).immutableCopy();
-            Set<String> added = Sets.difference(newCurrentSchemas, schemas.keySet()).immutableCopy();
-
-            for (String deletedSchema : deleted) {
-                try {
-                    schemas.remove(deletedSchema).close();
-                } catch (Exception e) {
-                    LOGGER.error(e.getMessage(), e);
-                }
-            }
-            for (String addedSchema : added) {
-                schemas.put(addedSchema, getCustomSchemaInfo(addedSchema));
+        Sets.SetView<String> nonBuiltInSchemas = Sets.difference(schemas.keySet(), builtInSchemas.keySet());
+        Set<String> deleted = Sets.difference(nonBuiltInSchemas, newCurrentSchemas).immutableCopy();
+        Set<String> added = Sets.difference(newCurrentSchemas, schemas.keySet()).immutableCopy();
```

Comment:
as this is no longer synchronized the different keySet() calls could provide different results?